### ISSUE TITLE
certbot renew --dry-run doesn't use ACME v2

### ISSUE THREAD

**Issue by mnordhoff** [certbot/certbot#5717]
## My operating system is (include version):

Ubuntu 16.04 (amd64)

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):

certbot-auto (0.22.0)

## I ran this command and it produced this output:

`$ sudo -H certbot certonly --manual --manual-auth-hook "/opt/certbot_dns/certbot_dns.py --auth --config /opt/certbot_dns/certbot_dns.ini --route-53" --manual-cleanup-hook "/opt/certbot_dns/certbot_dns.py --cleanup --config /opt/certbot_dns/certbot_dns.ini --route-53" --manual-public-ip-logging-ok --preferred-challenges dns-01 --server https://acme-staging-v02.api.letsencrypt.org/directory -d letsencrypt.mattnordhoff.info -d '*.letsencrypt.mattnordhoff.info'`

That's my not-open-source manual auth plugin for Route 53 DNS validation.

The certificate was successfully issued.

    $ sudo -H certbot renew --dry-run
    Saving debug log to /var/log/letsencrypt/letsencrypt.log

    [snip]

    -------------------------------------------------------------------------------
    Processing /etc/letsencrypt/renewal/letsencrypt.mattnordhoff.info.conf
    -------------------------------------------------------------------------------
    Cert not due for renewal, but simulating renewal for dry run
    Plugins selected: Authenticator manual, Installer None
    Renewing an existing certificate
    Attempting to renew cert (letsencrypt.mattnordhoff.info) from /etc/letsencrypt/renewal/letsencrypt.mattnordhoff.info.conf produced an unexpected error: The currently selected ACME CA endpoint does not support issuing wildcard certificates.. Skipping.

    [snip]

    The following certs could not be renewed:
      /etc/letsencrypt/live/letsencrypt.mattnordhoff.info/fullchain.pem (failure)
      /etc/letsencrypt/live/letsencrypt.mattnordhoff.net/fullchain.pem (failure)

    -------------------------------------------------------------------------------
    ** DRY RUN: simulating 'certbot renew' close to cert expiry
    **          (The test certificates below have not been saved.)

    The following certs were successfully renewed:
    [snip]

    The following certs could not be renewed:
      /etc/letsencrypt/live/letsencrypt.mattnordhoff.info/fullchain.pem (failure)
      /etc/letsencrypt/live/letsencrypt.mattnordhoff.net/fullchain.pem (failure)
    ** DRY RUN: simulating 'certbot renew' close to cert expiry
    **          (The test certificates above have not been saved.)
    -------------------------------------------------------------------------------
    2 renew failure(s), 0 parse failure(s)

(The `letsencrypt.mattnordhoff.net` certificate failed for the same reason.)

## Certbot's behavior differed from what I expected because:

Certbot tries to renew an ACME v2 staging wildcard certificate using the ACME v1 server, and naturally fails.

###### Logs are stored in `/var/log/letsencrypt` by default. Feel free to redact domains, e-mail and IP addresses as you see fit.

https://mn0.us/vPfhms7t_h2YLEaWk_AHaGK/letsencrypt.log (other certificates snipped)

I don't know if this is a known issue or expected behavior, but I didn't see anything. :confused:

**Comment by noci2012** [comment]
+1, same experience, --dry-run doesn't accept wildcard certificates for certbot 0.22

**Comment by MrBMT** [comment]
Can confirm the same behaviour with 0.22.0 on CentOS 7.

I used this command initially which worked to generate the wildcard certificate:
`certbot certonly --manual --manual-auth-hook '/etc/acme-dns/acme-dns-certbot-hook -config /etc/acme-dns/acme_dns.json' --manual-public-ip-logging-ok --no-eff-email --server https://acme-v02.api.letsencrypt.org/directory --preferred-challenges dns -d *.mydomain.co.uk
`

However doing certbot renew --dry-run also results in it using the wrong endpoint:
```
Starting new HTTPS connection (1): acme-staging.api.letsencrypt.org
Renewing an existing certificate
Attempting to renew cert (mydomain.co.uk-0001) from /etc/letsencrypt/renewal/mydomain.co.uk-0001.conf produced an unexpected error: The currently selected ACME CA endpoint does not support issuing wildcard certificates.. Skipping.
```

`/etc/letsencrypt/renewal/mydomain.co.uk-0001.conf` contains the correct endpoint listed as server, however this is ignored when `--dry-run` is used it seems:

```
# renew_before_expiry = 30 days
version = 0.22.0
archive_dir = /etc/letsencrypt/archive/mydomain.co.uk-0001
cert = /etc/letsencrypt/live/mydomain.co.uk-0001/cert.pem
privkey = /etc/letsencrypt/live/mydomain.co.uk-0001/privkey.pem
chain = /etc/letsencrypt/live/mydomain.co.uk-0001/chain.pem
fullchain = /etc/letsencrypt/live/mydomain.co.uk-0001/fullchain.pem

# Options used in the renewal process
[renewalparams]
account = MYACCOUNTKEY
manual_public_ip_logging_ok = True
manual_auth_hook = /etc/acme-dns/acme-dns-certbot-hook -config /etc/acme-dns/acme_dns.json
server = https://acme-v02.api.letsencrypt.org/directory
authenticator = manual
installer = None
pref_challs = dns-01,
```

**Comment by ohemorange** [comment]
Thanks for reporting this. There's currently no good way to handle this, as `--dry-run` updates the `--server`  contents to `acme-staging`. We'll definitely have a fix for this when we [update the default](https://github.com/certbot/certbot/issues/5369) to ACMEv2, and possible before then.

**Comment by ohemorange** [comment]
This will be fixed by #5732.
